sdk/core/azure-core-amqp/inc/azure/core/amqp/common/global_state.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright(c) Microsoft Corporation.All rights reserved.
+// SPDX - License - Identifier : MIT
+
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <list>
+#include <mutex>
+#include <thread>
+
+namespace Azure { namespace Core { namespace _internal { namespace Amqp { namespace Common {
+  namespace _detail {
+
+    /**
+     * uAMQP and azure-c-shared-util require that the platform_init and platform_uninit functions be
+     * called before using the various API functions.
+     *
+     * The GlobalState class maintains a singleton static local variable using [static local
+     * variables](https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables),
+     * also known as "Magic Statics". This ensures that we initialize only once.
+     *
+     */
+
+    class GlobalState {
+    public:
+      GlobalState();
+      ~GlobalState();
+
+      static GlobalState* GlobalStateInstance()
+      {
+        static auto globalState = new GlobalState();
```

Comment:
Should you create without the `new` and then return a pointer to it?

Reply:
Nobody ever deletes the globalState. And creating the globalState has to be done with a pointer, otherwise it's not a magic static (the whole trick here is the assignment at line 32 which is guaranteed to be atomic.